src/objects/Player.gd

```diff
@@ -173,13 +176,13 @@ remote func ive_been_hit():
 	is_vulnerable = false
 
 
-func shoot():
+func shooting_sound():
 	var stream_player := shooting.get_children()[rand_range(0, shooting.get_child_count())] as AudioStreamPlayer3D
 	stream_player.play()
-	hitscan.set_enabled(true)
-	hitscan.force_raycast_update()
-	if hitscan.is_colliding():
-		var hit := hitscan.get_collider()
-		if hit.has_method("on_raycast_hit"):
-			hit.on_raycast_hit(Multiplayer.get_player_id())
-	hitscan.set_enabled(false)
+
+
+func _on_Hurtbox_body_entered(body:Node) -> void:
+	if body.is_in_group("Arrow") and body.archer != self:
+		if is_vulnerable:
+			rpc("ive_been_hit")
+			ive_been_hit()
```

Comment:
We can instead make `ive_been_hit` use `remotesync` and get rid of the call to `ive_been_hit()` here. The RPC will also call it on ourself